include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -2124,6 +2333,189 @@ __parallel_partial_sort(oneapi::dpl::__internal::__device_backend_tag __backend_
     return __parallel_partial_sort_impl(__backend_tag, ::std::forward<_ExecutionPolicy>(__exec), __buf.all_view(),
                                         __partial_merge_kernel<decltype(__mid_idx)>{__mid_idx}, __comp);
 }
+
+//------------------------------------------------------------------------
+// reduce_by_segment - sync pattern
+//------------------------------------------------------------------------
+
+// TODO: The non-identity fallback path of reduce-by-segment must currently be implemented synchronously due to the
+// inability to create event dependency chains across separate parallel pattern calls. If we ever add support for
+// cross parallel pattern dependencies, then we can implement this as an async pattern.
```

Comment:
```suggestion
//------------------------------------------------------------------------
// reduce_by_segment - sync pattern
//
// TODO: The non-identity fallback path of reduce-by-segment must currently be implemented synchronously due to the
// inability to create event dependency chains across separate parallel pattern calls. If we ever add support for
// cross parallel pattern dependencies, then we can implement this as an async pattern.
//------------------------------------------------------------------------
```

Suggest that we move the comment about the synchronousity of the fallback up because its current placements looks like it's commenting on `__reduce1_wrapper` instead of the pattern as a whole.